src/test/java/spoon/test/reference/VariableAccessTest.java

```diff
@@ -146,6 +147,24 @@ public void testReferences() throws Exception {
 		System.out.println(make);
 	}
 
+	@Test
+	public void testReferenceToLocalVariableDeclaredInLoop() {
+		final class CtLocalVariableReferenceScanner extends CtScanner {
+			@Override
+			public <T> void visitCtLocalVariableReference(
+					final CtLocalVariableReference<T> reference) {
+				assertNotNull(reference.getDeclaration());
```

Comment:
add an assert on reference.getDeclaration().getSimpleName()? and maybe on getType()?